### PR TITLE
feat(cli): main interface

### LINKED ISSUE #35: Create a CLI to wrap the framework-core
The framework-core library should have a command-line interface so users can more easily integrate it with other environments just by downloading the node package and executing it i.e. GitHub Actions.

Given the bash command `code-suggester`, create arguments for:
- upstream-repo
- upstream-owner
- pr-description
- pr-title
- branch-name
- force
- files
- git-dir

as discussed in [go/auto-pr-framework](go/auto-pr-framework)

### PR DESCRIPTION
**Pull request by TomKristie**
Create a CLI interface to interact with the framework-core
It:
- takes arguments and maps them to user PR options
- takes a git directory and generates a change set Map from the git diff items
- initializes octokit and a logger
- invokes framework-core

- includes core-logic testing

Fixes #35

### CODE REVIEW

File: src/bin/code-suggester.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        'Required. The location of any un-tracked changes that should be made into a pull request. Files in the .gitignore are ignored.',
+      type: 'string',
+      demandOption: true,
+    },
+  })
+  .check(argv => {
+    for (const key in argv) {
+      if (typeof argv[key] === 'string' && !argv[key]) {
+        throw Error(
+          `String parameters cannot be provided empty values. Parameter \'${key}\' was given an empty value\'`
```
Comment (bcoe) [inline src/bin/code-suggester.ts:99]: would be nice if yargs did this for you. I like this approach you've taken.

File: src/bin/handle-git-dir-change.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import {readFile} from 'fs';
+import * as path from 'path';
+
+// data about a file in a git directory
+interface GitFileData {
+  path: string;
+  fileData: FileData;
+}
+
+// See https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-git-diff-filesltpatterngt82308203
```
Comment (bcoe) [inline src/bin/handle-git-dir-change.ts:27]: great use of a comment, It's really useful to link the canonical source of magic values like this.

File: test/git-dir-handler.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+// .null triggers ts-lint failure, but is valid chai
+describe('git directory diff output to git file data + content', () => {
+  const relativeGitDir = '/fixtures/some/git/dir';
+  const absoluteGitDir = __dirname + relativeGitDir;
+  const sandbox = sinon.createSandbox();
+  afterEach(() => {
+    // undo all changes
+    sandbox.restore();
+  });
+  it('the git diff parser does not read from file if the file status is deleted and uses the old mode', async () => {
```
Comment (bcoe) [inline test/git-dir-handler.ts:43]: pedantic-nit: I  usually  try to make  these read like sentences.

```
it does not read from a file if the file status is deleted and uses the old mode
```

File: test/git-dir-handler.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+});
+
+describe('parse all git diff output', () => {
+  const testDir = __dirname + '/fixtures';
+  const sandbox = sinon.createSandbox();
+  afterEach(() => {
+    // undo all changes
+    sandbox.restore();
+  });
+  it('populates change object with everything from a diff output', async () => {
```
Comment (bcoe) [inline test/git-dir-handler.ts:124]: these tests read very well,  and show a deep understanding of the git format 👍

File: src/bin/workflow.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    const octokit = new Octokit({auth: process.env.ACCESS_TOKEN});
+    switch (yargs.argv._[0]) {
+      case CREATE_PR_COMMAND:
+        changes = await git.getChanges(yargs.argv['git-dir'] as string);
+        break;
+      default:
+        // yargs should have caught this.
+        throw Error(`Unhandled command detected: ${yargs.argv._[0]}`);
+    }
+    await createPullRequest(octokit, changes, options, logger);
```
Comment (chingor13) [inline src/bin/workflow.ts:47]: This probably belongs in the CREATE_PR_COMMAND case

**Comment by TomKristie** [comment]
Commit history is in a kerfuffle because of rebasing.

**Pull request closed without merging**